third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -296,13 +296,21 @@ impl<'a> Transformer<'a> {
             },
             Call(id, dests, op, srcs, ai) => {
                 use Operation::*;
-                match &op {
-                    Function(..) => {
-                        let new_srcs = self.copy_args_if_needed(code_offset, id, srcs);
-                        self.check_and_emit_bytecode(code_offset, Call(id, dests, op, new_srcs, ai))
-                    },
-                    _ => self.check_and_emit_bytecode(code_offset, bc.clone()),
-                }
+                let new_bc = if op.can_take_non_ref_primitive_as_args() {
```

Comment:
You should pass `op` into `copy_args_if_needed()` and do this logic there.  This function already has enough going on.

Reply:
done